pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -1391,6 +1429,9 @@ private void processOpSendMsg(OpSendMsg op) {
                 batchMessageAndSend();
             }
             pendingMessages.put(op);
+            if (op.msg != null) {
+                lastSequenceIdPushed = op.sequenceId;
```

Comment:
1. I think we need to keep track highestSequenceId, no?
2. We should only assign the highest sequence id, no?

```
lastSequenceIdPushed = Math.max(lastSequenceIdPushed, op.lastSequenceIdPushed);
```